File: src/solutions/Engine/BO/FlowControl/Device.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Engine.BL.Actuators2;

namespace Engine.BO.FlowControl
{
    public class Device : BaseBO
    {
        public static FlowBL bl = new ();

        public string? Name { get; set; }        
        public string? Ip { get; set; }
        public string? Model { get; set; }
        public bool IsActive { get; set; }

        [JsonPropertyName("last_update")]
        public DateTime? LastUpdate { get; set; }
        public Flow? Flow  => bl.GetFlowByDevice((int)Id);
    }
}

File: src/FlowClient/src/interfaces/catalog/Device.ts
import { Flow } from "./../api/Api"

export interface Device {
  name: string;
  ip: string;
  model: string;
  isActive: boolean;
  last_update: Date;
  id: number;
  flow?: Flow
}

export function parseDevice(d: Device) : Device {
  if (typeof d.last_update == "string") {
    d.last_update = new Date(d.last_update);
  }

  return d;
}

export function parseDevices(dList: Device[]) : Device[] {
  return dList.map(d => parseDevice(d));
}
